staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/controller.go

```diff
@@ -187,30 +200,52 @@ func (c *celAdmissionController) Validate(
 	c.mutex.RLock()
 	defer c.mutex.RUnlock()
 
-	var allDecisions []PolicyDecisionWithMetadata = nil
+	var deniedDecisions []policyDecisionWithMetadata
+
+	addConfigError := func(err error, definition *v1alpha1.ValidatingAdmissionPolicy, binding *v1alpha1.ValidatingAdmissionPolicyBinding) {
+		// we always default the FailurePolicy if it is unset and validate it in API level
+		var policy v1alpha1.FailurePolicyType
+		if definition.Spec.FailurePolicy == nil {
+			policy = v1alpha1.Fail
+		} else {
+			policy = *definition.Spec.FailurePolicy
+		}
 
-	addConfigError := func(err error, definition PolicyDefinition, binding PolicyBinding) {
-		wrappedError := fmt.Errorf("configuration error: %w", err)
-		switch p := definition.GetFailurePolicy(); p {
-		case Ignore:
-			klog.Info(wrappedError)
+		// apply FailurePolicy specified in ValidatingAdmissionPolicy, the default would be Fail
+		switch policy {
+		case v1alpha1.Ignore:
+			if binding == nil {
+				klog.Infof("ignored ValidatingAdmissionPolicy %s failure, due to FailurePolicy=Ignore. Error: %w", definition.Name, err)
+			} else {
+				klog.Infof("ignored ValidatingAdmissionPolicy %s failure for binding %s, due to FailurePolicy=Ignore. Error: %w", definition.Name, binding.Name, err)
+			}
 			return
-		case Fail:
-			allDecisions = append(allDecisions, PolicyDecisionWithMetadata{
-				PolicyDecision: PolicyDecision{
-					Kind:    Deny,
-					Message: wrappedError.Error(),
+		case v1alpha1.Fail:
+			var message string
+			if binding == nil {
+				message = fmt.Errorf("failed to configure policy: %w", err).Error()
+			} else {
+				message = fmt.Errorf("failed to configure binding: %w", err).Error()
+			}
+			deniedDecisions = append(deniedDecisions, policyDecisionWithMetadata{
+				policyDecision: policyDecision{
+					kind:    deny,
+					message: message,
 				},
-				Definition: definition,
-				Binding:    binding,
+				definition: definition,
+				binding:    binding,
 			})
 		default:
-			utilruntime.HandleError(fmt.Errorf("unrecognized failure policy: '%v'", p))
+			utilruntime.HandleError(fmt.Errorf("unrecognized failure policy: '%v'", policy))
 		}
 	}
 	for definitionNamespacedName, definitionInfo := range c.definitionInfo {
 		definition := definitionInfo.lastReconciledValue
-		if !definition.Matches(a) {
+		matches, matchKind, err := c.validatorCompiler.DefinitionMatches(a, o, definition)
+		if err != nil {
+			return err
```

Comment:
why is this not a configuration error? if this is for a policy that is failureMode=Ignore, this will turn a match error to a fatal error, right?